src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/LSPExtendWordSelectionHandler.java

```diff
@@ -33,31 +33,23 @@
 /**
  * Implementation of the IDE's extendWordSelectionHandler EP for LSP4IJ files against textDocument/selectionRange.
  */
-public class LSPExtendWordSelectionHandler implements ExtendWordSelectionHandler {
+public class LSPExtendWordSelectionHandler extends AbstractLSPExtendWordSelectionHandler {
 
     @Override
     public boolean canSelect(@NotNull PsiElement element) {
-        if (!element.isValid()) {
+        if (!super.canSelect(element)) {
             return false;
         }
 
-        PsiFile psiFile = element.getContainingFile();
-        if ((psiFile == null) || !psiFile.isValid()) {
-            return false;
-        }
-
-        Project project = psiFile.getProject();
-        if (project.isDisposed()) {
-            return false;
-        }
+        // These should all be safely non-null now
```

Comment:
And that means that if `super.canSelect()` returns true, we know that we have good `Project`, `PsiFile`, and `VirtualFile` data to work with.